src/Tools/dotnet-monitor/ConfigurationJsonWriter.cs

```diff
@@ -217,25 +220,64 @@ private void ProcessSection(IConfigurationSection section, bool includeChildSect
                     {
                         if (child.GetChildren().Any())
                         {
-                            ProcessChildren(child, includeChildSections, redact);
+                            ProcessChildren(child, includeChildSections, redact,  showSources: showSources);
                         }
                         else
                         {
                             WriteValue(child.Value, redact);
+
+                            string comment = GetConfigurationProvider(child, showSources);
+
+                            if (comment.Length > 0)
```

Comment:
Can you put a TODO regarding the comment ordering, explaining why it's after the value instead of before?